### PR TITLE
[DEV-15631] Add support for specifying remote cache key timeout

### PR DESCRIPTION
**Pull request by bal2ag**

### CODE REVIEW

File: modeldict/base.py [hunk 1]
```diff
@@ -20,6 +25,9 @@ def __init__(self, cache=cache, timeout=30):
         self.remote_cache_key = cls_name
         self.remote_cache_last_updated_key = '%s.last_updated' % (cls_name,)
 
+        if not remote_timeout == DefaultRemoteTimeout:
```
Comment (Jaidan) [inline modeldict/base.py]: I would probably use `is` here since we really want to be doing an identity check.

File: modeldict/base.py [hunk 2]
```diff
@@ -219,3 +231,8 @@ def _cleanup(self, *args, **kwargs):
         # We set _last_updated to a false value to ensure we hit the
         # last_updated cache on the next request
         self._last_checked_for_remote_changes = 0.0
+
+    def _cache_set_kwargs(self):
```
Comment (Jaidan) [inline modeldict/base.py]: I think with would be a good place to use `@property`

**Review by Jaidan — CHANGES_REQUESTED** [review]

File: modeldict/base.py (hunk 1 above)
Comment (bal2ag) [reply modeldict/base.py]: Fixed

File: modeldict/base.py (hunk 2 above)
Comment (bal2ag) [reply modeldict/base.py]: Fixed

**Comment by bal2ag** [comment]
@Jaidan comments addressed

**Review by Jaidan — APPROVED** [review]

**Comment by Jaidan** [comment]
We should probably push this out to a staging environment before production

**Pull request merged**